system_setup/server/controllers/locale.py

```diff
@@ -30,3 +32,65 @@ def start(self):
                           .format(self.model.selected_language))
 
         super().start()
+
+    async def _run_locale_support_cmds(self, lang: str):
+        """ Final commands to run when a valid language is POST'ed. """
+        cmds = [["locale-gen"],
+                ["update-locale", "LANG={}".format(lang)]]
+        env = os.environ.copy()
+
+        cp = await arun_command(["check-language-support", "-l", lang[0:5]],
+                                env=env)
+        if cp.returncode == 0:
+            packages = cp.stdout.strip('\n').split(' ')
+            cmds.append(["apt", "install", "-y"] + packages)
+
+        if self.app.opts.dry_run:
+            for cmd in cmds:
+                log.debug('Would run: ' + ' '.join(cmd))
+        else:
+            for cmd in cmds:
+                cp = await arun_command(cmd, env=env)
+                if cp.returncode:
+                    log.debug('Command \"%s\" failed with return code %d',
+                              cp.args, cp.returncode)
+
+    async def POST(self, data: str):
+        if data == self.autoinstall_default or data == os.environ.get("LANG"):
+            await super().POST(data)
+            return
+
+        fileContents: str
+        fname = "locale.gen"
+        env = os.environ.copy()
+        if self.app.opts.dry_run:
+            # For debug purposes.
+            fname = ".subiquity/" + fname
+            await arun_command(['cp', '/etc/locale.gen', '.subiquity/'],
```

Comment:
I think there is already an etc in .subiquity, so you can put it here. Also, hem, there are some method in python to copy a file if this will stay in the end.